### DISCUSSION TITLE
Managing conflicting STAC extensions

### DISCUSSION THREAD

**Discussion by fmigneault** [radiantearth/stac-spec#1285] [General STAC Q&A]
Currently, there are (to my knowledge) 3 variations of `cmip6` extension:

- https://github.com/TomAugspurger/cmip6
- https://github.com/dchandan/stac-extension-cmip6
  (fork from the previous one with fixes, used by https://github.com/crim-ca/stac-populator/blob/master/STACpopulator/extensions/cmip6.py)
- https://github.com/stac-extensions/cmip6 
  (new from ~1month ago)

This raises a few questions:
1. How to handle conflicting names? 
   - Should `stac_extensions` used a namespace mapping (eg: `{"cmip6": "<uri-to-extension>"}`) to make the references more explicit?
   - If yes, how to "remap" the namespace in the property names to allow multiple extensions simultaneously?
   - If no, how to pick one over another?

2. How to best combine similar efforts?
   - How to provide better communication of intentions to develop new extensions, such that people can flag duplicate work earlier?

FYI @mishaschwartz @dchandan @rhysrevans3 @huard @m-mohr

**Comment by m-mohr** [comment]
None of them is correctly released, i.e. there is no schema which you can validate against. I think they should all be merged together...

Usually, having multiple similar extensions shouldn't (technically) be a big problem because the Extension identifiers are different so you can detect to which extensions fields belong. So you don't have issues with overlapping prefixes.

But of course ideally the community would bring them together and work on a single extension. One issue is that people don't understand that they should make their extension better known (submit to stac-extensions.github.io) so that issues like with cmip6 ideally don't occur because people find the existing extension...

For now I'd just raise issues in all repos with the same text, asking to work with the others to combine into a single extension.

**Reply by dchandan** [reply]
Could you please clarify what you mean when you say there is no schema to validate against? All the extensions do have a schema (for example, https://raw.githubusercontent.com/dchandan/stac-extension-cmip6/v1.0.0/json-schema/schema.json). This schema can be used to validate the STAC items created using the extension. Am I missing something?

**Reply by fmigneault** [reply]
Ideally, it would be directly served by some endpoint hosted on GitHub pages. 
Typically, that would be https://dchandan.github.io/stac-extension-cmip6/v1.0.0/schema.json if hosted under your org-user/repository. 

Currently, the readme points at https://stac-extensions.github.io/cmip6/v1.0.0/schema.json, (where https://github.com/stac-extensions/cmip6 would actually be published), which is invalid and potentially very misleading.

**Reply by m-mohr** [reply]
Yeah, the extension points to https://stac-extensions.github.io/cmip6/v1.0.0/schema.json which doesn't resolve so for me it looks like there is no schema.

**Comment by rhysrevans3** [comment]
@fmigneault thanks for bring this up I didn't actually realise that there were any existing implementations. I agree with @m-mohr it would be great if we could collaborate and merge these extensions. I've been waiting for colleagues to have time to check I've include the right facets in the [stac-entensions/cmip6](https://github.com/stac-extensions/cmip6) repo as I don't really have the domain knowledge. 

@fmigneault out of interest do you know if either of the other CMIP6 extensions are being used in any public catalogs?

**Reply by m-mohr** [reply]
None of the extensions has a public schema it seems, so no deployment of such metadata would actually validate. So I doubt there are serious implementations.

While merging the extensions, make sure to check for existing fields. It looks like some of the fields could be replaced by other fields from core or other extensions.

**Reply by fmigneault** [reply]
@rhysrevans3 
I only know of https://github.com/dchandan/stac-extension-cmip6 that is used by https://github.com/crim-ca/stac-populator, which itself is used by https://github.com/bird-house/birdhouse-deploy to manage the nodes listed in https://marbleclimate.com/node.html.

I think that https://github.com/dchandan/stac-extension-cmip6 mostly applied CMIP6 vocabulary fixes over https://github.com/TomAugspurger/cmip6, so if those fixes can be integrated into https://github.com/stac-extensions/cmip6, we could mark both of these as deprecated in favor of the published one under `stac-extensions` (after release of the JSON-schema of course).
